pkg/monitor/event.go

```diff
@@ -8,175 +8,155 @@ import (
 	"time"
 
 	"github.com/openshift/origin/pkg/monitor/monitorapi"
-
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
 )
 
 func startEventMonitoring(ctx context.Context, m Recorder, client kubernetes.Interface) {
 	reMatchFirstQuote := regexp.MustCompile(`"([^"]+)"( in (\d+(\.\d+)?(s|ms)$))?`)
 
-	go func() {
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			default:
-			}
-
-			// filter out events written "now" but with significantly older start times (events
-			// created in test jobs are the most common)
-			significantlyBeforeNow := time.Now().UTC().Add(-15 * time.Minute)
+	// filter out events written "now" but with significantly older start times (events
+	// created in test jobs are the most common)
+	significantlyBeforeNow := time.Now().UTC().Add(-15 * time.Minute)
 
-			events, err := client.CoreV1().Events("").List(ctx, metav1.ListOptions{Limit: 1})
-			if err != nil {
-				continue
+	listWatch := cache.NewListWatchFromClient(client.CoreV1().RESTClient(), "events", "", fields.Everything())
+	customStore := &cache.FakeCustomStore{
```

Comment:
I haven't looked in a while.  What does a delete look like?

Reply:
Old code did not consider deletes at all, and that seemed intentional and correct to me as I don't think it matters when they get cleaned up to us. Let me know if there's anything I'm missing there.